app/src/main/java/com/nextcloud/talk/serverstatus/ServerStatusRepositoryImpl.kt

```diff
@@ -0,0 +1,45 @@
+/*
+ * Nextcloud Talk - Android Client
+ *
+ * SPDX-FileCopyrightText: 2025 Sowjanya Kota <sowjanya.kch@gmail.com>
+ * SPDX-License-Identifier: GPL-3.0-or-later
+ */
+
+package com.nextcloud.talk.serverstatus
+
+import com.nextcloud.talk.api.NcApiCoroutines
+import com.nextcloud.talk.data.user.model.User
+import com.nextcloud.talk.utils.ApiUtils
+import com.nextcloud.talk.utils.database.user.CurrentUserProviderNew
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import javax.inject.Inject
+
+class ServerStatusRepositoryImpl @Inject constructor(
+    private val ncApiCoroutines: NcApiCoroutines,
+    private val currentUserProvider: CurrentUserProviderNew
+) : ServerStatusRepository {
+    private val _currentUser = currentUserProvider.currentUser.blockingGet()
+    val currentUser: User = _currentUser
+
+    private var _isServerReachable: MutableStateFlow<Boolean> = MutableStateFlow(false)
+    override val isServerReachable: StateFlow<Boolean>
+        get() = _isServerReachable.asStateFlow()
+
+    override suspend fun getServerStatus() {
+        serverStatus()
+    }
+
+    @Suppress("Detekt.TooGenericExceptionCaught")
+    private suspend fun serverStatus() {
+        val baseUrl = currentUser.baseUrl
+        val url = baseUrl + ApiUtils.getUrlPostfixForStatus()
+        _isServerReachable.value = try {
+            ncApiCoroutines.getServerStatus(url)
+            true
+        } catch (e: Exception) {
```

Comment:
check if e is a `CancellationException`, if so throw it. [The CancellationException](https://kotlinlang.org/api/core/kotlin-stdlib/kotlin.coroutines.cancellation/-cancellation-exception/) is a special kind of exception that is needed for proper coroutine behavior. Will not throwing it cause your PR to fail? Maybe not for now, but it's considered [good practice.](https://stackoverflow.com/questions/76259793/is-it-necceary-to-rethrow-the-cancellationexception-in-kotlin)